spec/syncSpec.js

```diff
@@ -204,14 +250,18 @@ function updateRecord(uid) {
   });
 }
 
-function doUpdate() {
-  return function(res) {
-    return new Promise(function(resolve, reject) {
-      $fh.sync.doUpdate(datasetId, res.uid, updateData, function() {
-        resolve(res);
-      }, function(err) {
-        reject(err);
-      });
-    });
-  };
-}
+/**
+ * Iterate through the elements of an object and return the first element
+ * which returns `true` for the `evaluator` function argument.
+ *
+ * @param {Object} obj - The object to search.
+ * @param {Function} test - The function to test each element with.
+ * @returns {any} - The first element in `obj` which passed `test`.
+ */
+function searchObject(obj, test) {
```

Comment:
@aidenkeating I've just run this locally and took a quick look. I was wondering is you could `filter` instead of the searchObject? But that is not a biggie and we can revisit later. I wanted to give a 👍  as I'll probably not be working today and so that you are not held up by this.